src/plugins/map/MQ2MapCommands.cpp

```diff
@@ -1364,6 +1580,387 @@ static void DrawMapSettings_Colors()
+	{
+		if (!IsMapLocsStyleOverridden())
+		{
+			gOverrideMapLocParams.circleColor.Blue = static_cast<uint8_t>(radiusColor.Value.z * 255);
+			gOverrideMapLocParams.circleColor.Green = static_cast<uint8_t>(radiusColor.Value.y * 255);
+			gOverrideMapLocParams.circleColor.Red = static_cast<uint8_t>(radiusColor.Value.x * 255);
+			gOverrideMapLocParams.circleColor.Alpha = 255;
+		}
+
+		gDefaultMapLocParams.circleColor.Blue = static_cast<uint8_t>(radiusColor.Value.z * 255);
+		gDefaultMapLocParams.circleColor.Green = static_cast<uint8_t>(radiusColor.Value.y * 255);
+		gDefaultMapLocParams.circleColor.Red = static_cast<uint8_t>(radiusColor.Value.x * 255);
+		gDefaultMapLocParams.circleColor.Alpha = 255;
+
+		WritePrivateProfileInt("MapLoc", "RadiusRed", gDefaultMapLocParams.circleColor.Red, INIFileName);
+		WritePrivateProfileInt("MapLoc", "RadiusGreen", gDefaultMapLocParams.circleColor.Green, INIFileName);
+		WritePrivateProfileInt("MapLoc", "RadiusBlue", gDefaultMapLocParams.circleColor.Blue, INIFileName);
+		UpdateDefaultMapLocInstances();
+		regenerate = true;
+	}
+
+	ImGui::Separator();
+
+	if (regenerate)
+	{
+		MapClear();
+		MapGenerate();
+	}
+}
+
+static bool IsAnyMapLocSelected()
+{
+	for (size_t i = 0; i < gMapLocTemplates.size(); i++)
+	{
+		if (gMapLocTemplates[i]->GetMapLoc()->m_isSelected)
+		{
+			return true;
+		}
+	}
+
+	return false;
+}
+
+static void DeleteSelectedMapLocs()
+{
+	for (size_t i = 0; i < gMapLocTemplates.size(); i++)
+	{
+		if (gMapLocTemplates[i]->GetMapLoc()->m_isSelected)
+		{
+			DeleteMapLoc(gMapLocTemplates[i--]->GetMapLoc());
+		}
+	} 
+}
+
+static void ResetSelectedMapLocsToDefault()
+{
+	for (size_t i = 0; i < gMapLocTemplates.size(); i++)
+	{
+		if (gMapLocTemplates[i]->GetMapLoc()->m_isSelected)
+		{
+			MapObjectMapLoc* thisMapLoc = gMapLocTemplates[i]->GetMapLoc();
+			gMapLocTemplates[i]->SetCreatedFromDefaults(true);
+			gMapLocTemplates[i]->UpdateFromParams(gDefaultMapLocParams);
+		}
+	}
+}
+
+static void ApplyOverridesToSelected(MapLocParams params)
+{
+	for (size_t i = 0; i < gMapLocTemplates.size(); i++)
+	{
+		if (gMapLocTemplates[i]->GetMapLoc()->m_isSelected)
+		{
+			MapObjectMapLoc* thisMapLoc = gMapLocTemplates[i]->GetMapLoc();
+			gMapLocTemplates[i]->SetCreatedFromDefaults(false);
+			gMapLocTemplates[i]->UpdateFromParams(params);
+		}
+	}
+}
+
+static void DrawMapSettings_MapLocs()
+{
+	bool regenerate = false;
+	bool disable = false;
+
+	if (gMapLocTemplates.size() < 1)
+	{
+		ImGui::BeginDisabled();
+		disable = true;
+	}
+	if (ImGui::Button("Clear All Map Locs"))
+	{
+		ImGui::OpenPopup("Delete?");
+	}
+	if (disable)
+	{
+		ImGui::EndDisabled();
+		disable = false;
+	}
+
+	ImVec2 center = ImGui::GetMainViewport()->GetCenter();
+	ImGui::SetNextWindowPos(center, ImGuiCond_Appearing, ImVec2(0.5f, 0.5f));
+	if (ImGui::BeginPopupModal("Delete?", NULL, ImGuiWindowFlags_AlwaysAutoResize))
+	{
+		ImGui::Text("Are you sure you want to delete all map locs?\n\n");
```

Comment:
ImGui::NewLine() instead of making a multiline text

Reply:
I think I stole this from the imgui example haha, but I'll fix it